src/sbm.jl

```diff
@@ -916,6 +921,14 @@ function update_until_recharge(sbm::SBM, config)
             sbm.waterfrac[i] * sbm.potential_evaporation[i],
         )
 
+        # Update land waterlevel
+        waterlevel_land = sbm.waterlevel_land[i] - ae_openw_l
+
+        if do_surface_water_infiltration
+            # Add land waterlevel to infiltration
+            avail_forinfilt += waterlevel_land
```

Comment:
The amount of surface water that can infiltrate needs to be scaled by the land fraction. We use a similar approach for runoff and evaporation from open water (using `waterfrac`).